src/commands/synthetics/interfaces.ts

```diff
@@ -211,8 +211,9 @@ interface BasicAuthCredentials {
   username: string
 }
 
-export interface TemplateContext extends NodeJS.ProcessEnv {
+export interface TemplateVariables {
```

Comment:
To prevent duplication between `TemplateVariables` and `reservedNames`, we might try something like this:
```typescript
const optionalTemplateVariables = ['DOMAIN', 'HASH', ...] as const
const mandatoryTemplateVariables = ['URL', ...] as const
type OptionalTemplateVariables = typeof optionalTemplateVariables[number]
type MandatoryTemplateVariables = typeof mandatoryTemplateVariables[number]
export type TemplateVariables = Partial<Record<OptionalTemplateVariables, string>> & Record<MandatoryTemplateVariables, string>

const invalidTemplateVariable: TemplateVariables = {
  HASH: 'hash'
}

const validTemplateVariables: TemplateVariables = {
  URL: 'url'
}
```